py3status/py3.py

```diff
@@ -40,6 +51,41 @@ def __repr__(self):
         return 'None'
 
 
+class HTTPResponse:
+    """
+    This class is like a simplified version of `urllib3.response.HTTPResponse`.
+
+    The HTTPResponse object provides status, data, and json attributes.
+
+    @TODO: Implement the header attribute
+    """
+
+    def __init__(self, resp_obj):
+        self._resp_obj = resp_obj
+        self._data = None
+        self._json = None
+
+    def __getattr__(self, name):
+        # If trying to access an attribute that doesn't exist in this class,
+        # try accessing that attribute from the response object.
+        return getattr(self._resp_obj, name)
+
+    @property
+    def status(self):
+        return self._resp_obj.getcode()
+
+    @property
+    def data(self):
+        if self._data is None:
+            self._data = self._resp_obj.read()
+        return self._data
+
+    def json(self):
+        if self._json is None:
+            self._json = json.loads(self.data.decode('utf-8'))
```

Comment:
I left `.data` as bytes to be consistent with all other libraries, since, by definition, the data is actually `bytes`.  That is actually how `urllib3` [shows how to get JSON data](http://urllib3.readthedocs.io/en/latest/user-guide.html#json-content).